src/quera_ahs_utils/plotting.py

```diff
@@ -29,34 +30,29 @@ def show_register(
             register (AtomArrangement): A given register
             blockade_radius (float): Default is 0. The blockade radius for the register.
             what_to_draw (str): Default is "bond". Either "bond" or "circle" to indicate the blockade region. 
-            show_atom_index (bool): Default is True. Choose if each atom's index is displayed over the atom itself in the resulting figure. 
-        
+            show_atom_index (bool): Default is True. Choose if each atom's index is displayed over the atom itself in the resulting figure.  
     """
-    filled_sites = [site.coordinate for site in register if site.site_type == SiteType.FILLED]
-    empty_sites = [site.coordinate for site in register if site.site_type == SiteType.VACANT]
-    
-    fig = plt.figure(figsize=(7, 7))
-    if filled_sites:
-        plt.plot(np.array(filled_sites)[:, 0], np.array(filled_sites)[:, 1], 'r.', ms=15, label='filled')
-    if empty_sites:
-        plt.plot(np.array(empty_sites)[:, 0], np.array(empty_sites)[:, 1], 'k.', ms=5, label='empty')
-    plt.legend(bbox_to_anchor=(1.1, 1.05))
-    
-    if show_atom_index:
-        for idx, site in enumerate(register):
-            plt.text(*site.coordinate, f"  {idx}", fontsize=12)
-    
-    if blockade_radius > 0 and what_to_draw=="bond":
-        for i in range(len(filled_sites)):
-            for j in range(i+1, len(filled_sites)):            
-                dist = np.linalg.norm(np.array(filled_sites[i]) - np.array(filled_sites[j]))
-                if dist <= blockade_radius:
-                    plt.plot([filled_sites[i][0], filled_sites[j][0]], [filled_sites[i][1], filled_sites[j][1]], 'b')
-                    
-    if blockade_radius > 0 and what_to_draw=="circle":
-        for site in filled_sites:
-            plt.gca().add_patch( plt.Circle((site[0],site[1]), blockade_radius/2, color="b", alpha=0.3) )
-        plt.gca().set_aspect(1)
+    
+    positions = np.array([site.coordinate for site in register])
+    is_filled = [site.site_type == SiteType.FILLED for site in register]
+    is_empty  = [site.site_type == SiteType.EMPTY for site in register]
+    
+    
+    if what_to_draw=="bond":
+        ax, G = visualize_UDG(positions = positions[is_filled,:],
```

Comment:
```suggestion
        ax, _ = visualize_UDG(positions = positions[is_filled,:],
```
If G is not used/returned, you can just assign it to `_`